src/design-system/components/checkbox/CheckBox.tsx

```diff
@@ -0,0 +1,74 @@
+import Checkbox, { CheckboxProps } from '@mui/material/Checkbox'
+import { FC, ReactNode, useState } from 'react'
+import Loader from '~/components/loader/Loader'
+
+import './CheckBox.scss'
+
+interface CheckBoxProps extends Omit<CheckboxProps, 'size'> {
+  variant: 'check' | 'middle'
+  label: ReactNode
+  labelPosition?: 'top' | 'bottom' | 'end'
+  color?: 'primary' | 'secondary' | 'error' | 'success'
+  size?: 'sm' | 'md' | 'lg'
+  loading?: boolean
+}
+
+const CheckBox: FC<CheckBoxProps> = ({
+  color = 'primary',
+  disabled = false,
+  label,
+  labelPosition = 'end',
+  loading = false,
+  variant = 'check',
+  size = 'md',
+  ...props
+}) => {
+  const [checked, setChecked] = useState(false)
```

Comment:
Add generic here.
It is a good practice to have generic type in useState hook, even if it has default value